src/main/resources/jenkins/plugins/openstack/compute/OpenStack.groovy

```diff
@@ -0,0 +1,89 @@
+package jenkins.plugins.openstack.compute
+
+import jenkins.plugins.openstack.compute.internal.DestroyMachine
+import jenkins.plugins.openstack.compute.internal.Openstack
+import org.openstack4j.model.compute.Server
+
+class OpenStack implements Serializable {
+
+    private org.jenkinsci.plugins.workflow.cps.CpsScript script
+
+    public OpenStack(org.jenkinsci.plugins.workflow.cps.CpsScript script) {
+            this.script = script
+    }
+
+    public Node server(String cloudName, String template) {
+        new Node(this, cloudName, template)
+    }
+
+    public static class Node implements Serializable {
+
+        private final OpenStack openstack
+        public final String cloudName
+        public final String template
+        private Server server = null
+
+        private Node(OpenStack openstack, String cloudName, String template) {
+            this.openstack = openstack
+            this.cloudName = cloudName
+            this.template = template
+        }
+
+        public void destroy() {
+            if (server != null) {
+                DestroyMachine dm
+                dm = new DestroyMachine(cloudName, server.getId())
+                dm.dispose()
+                server = null
+            } else {
+                //Cant destroy what isn't created
```

Comment:
The last commmit handles overprovisionig more than one node wit a boot() or bootUnmanaged() call. This code will suceed, booting two instances:

```def node1 = openstack.server("test", "o-RHEL-7")
node1.boot()    //scoped to current run
echo "n1 ${node1.address}"
node1.bootUnmanaged()   //Won't change the scope
echo "n1 ${node1.address}"   //Same address as before, as the node is the same
def node2 = openstack.server("test", "o-RHEL-7")
node2.bootUnmanaged()
echo "n2 ${node2.address}"
node2.destroy()
```